lib/puppet/pops/loader/ruby_legacy_function_instantiator.rb

```diff
@@ -60,4 +67,41 @@ def self.get_binding(loader_injected_arg)
     binding
   end
   private_class_method :get_binding
+
+  def self.assert_code(code_string, result)
+    ripped = Ripper.sexp(code_string)
```

Comment:
`Ripper#sexp` is an experimental API. Seems not good to rely on it for anything. Would `Ripper#lex` work instead?

Reply:
not really, then I would have to write a Ruby parser